core/src/base/mod.rs

```diff
@@ -37,15 +37,16 @@ pub struct Score {
 }
 
 impl Score {
-    pub fn calc_hash(path: impl AsRef<Path>) -> String {
-        format!(
-            "{:x}",
-            arklib::id::ResourceId::compute(
-                File::open(&path).unwrap().metadata().unwrap().len(),
-                path,
-            )
-            .crc32
-        )
+    pub fn calc_hash(path: impl AsRef<Path>) -> Result<String, std::io::Error> {
+        let file_metadata = std::fs::metadata(&path)?;
+        let ressource =
+            arklib::id::ResourceId::compute(file_metadata.len(), path).map_err(|_| {
+                std::io::Error::new(
+                    std::io::ErrorKind::InvalidInput,
+                    "Error computing RessourceId",
+                )
+            })?;
+        Ok(ressource.crc32.to_string())
```

Comment:
Actually, it should be `calc_id` returning composite id (crc32 and also filesize). CRC-32 is pretty weak hash function (the fastest I found) and using file size decreases ratio of collisions significantly.

`Ok(resource.to_string)` or something like `format!("{}-{}", resource.crc32, resource.size)` in the end should be good. We also can use `ResourceId` directly, it is marked as `Copy` in arklib.

If we need to pass size to arklib together with the path, then we should modify `compute` method.